node/pkg/watchers/sui/watcher.go

```diff
@@ -189,16 +180,67 @@ func (e *Watcher) inspectBody(logger *zap.Logger, body SuiResult) error {
 
 	if e.suiMoveEventType != *body.Type {
 		logger.Info("type mismatch", zap.String("e.suiMoveEventType", e.suiMoveEventType), zap.String("type", *body.Type))
-		return errors.New("type mismatch")
+		return nil
 	}
 
-	fields := *body.Fields
-	if (fields.ConsistencyLevel == nil) || (fields.Nonce == nil) || (fields.Payload == nil) || (fields.Sender == nil) || (fields.Sequence == nil) {
-		logger.Info("Missing required fields in event.")
+	fieldsMap := *body.Fields
```

Comment:
I'm wondering if there is a way to defer parsing this until after the above type check, that way we can keep the old parsing.

Reply:
Fixed.